src/components/Select2.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <section>
+    <div ref="select" class="select" @click="() => { toggleSelect() }">
+      <span class="select-holder">{{ selectedOption || '-- Select data source' }}</span>
```

Comment:
Keep the term of **data source** out of `Select2.vue`  because the UI/UX has no relation to data sources. This means the Vue component can be used exactly as it is for any other widgets.

```suggestion
      <span class="select-holder">{{ selectedOption || '-- Select an option' }}</span>
```